### PR TITLE
feature: add configuration for build docker image in server module

### PR DESCRIPTION
**Pull request by helloworlde**
### Ⅰ. Describe what this PR did
Add configuration for build Docker image in server module.

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
This PR can't add test cases in code. 

### Ⅳ. Describe how to verify it
1. Add `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` environment which docker account and password.
2. Execute `mvn clean package -P image` and then will push `latest` tag image in your docker hub.

### Ⅴ. Special notes for reviews

- Need to add two environment parameter in Travis: `REGISTRY_USERNAME` and `REGISTRY_PASSWORD`
- The namespace `seata.io` need to check if is belong to us, or need to change others.
- Add one profile for docker image named `image`
- There has two scene to build image: 
  - Use `image` profile, will build and upload the `latest` tag image.
  - Use `release` profile, will build and upload the `latest` and `${project.version}` tag images.

**And there has a very important thing is that we need change path of `registry.conf` and  `file.conf`, now if we want mount configuration volume to container, we need add not only `registry.conf` and `file.conf`, but also all files under `resources`. So move the two files to another folder maybe is a good choice**

### CODE REVIEW

File: server/src/main/java/io/seata/server/ParameterParser.java [hunk 1]
```diff
@@ -18,9 +18,18 @@
 import com.beust.jcommander.JCommander;
 import com.beust.jcommander.Parameter;
 import com.beust.jcommander.ParameterException;
-import io.seata.common.util.StringUtils;
 import io.seata.config.ConfigurationFactory;
 import io.seata.core.constants.ConfigurationKeys;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
```
Comment (zjinlei) [inline server/src/main/java/io/seata/server/ParameterParser.java]: recommended to extend the StringUtils and NumberUtils of seata

Comment (helloworlde) [reply server/src/main/java/io/seata/server/ParameterParser.java]: There is not `NumberUtils` exist in project, and `StringUtils` doesn't have `defaultIfBlank` method.
So used apache utility library.

**Comment by slievrly** [comment]
@helloworlde How to control the account permissions of docker.io?

**Comment by helloworlde** [comment]
> @helloworlde How to control the account permissions of docker.io?

As we discussed in Dingtalk group, use profile and environment.

File: server/pom.xml [hunk 2]
```diff
@@ -153,6 +153,72 @@
...
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>jib-maven-plugin</artifactId>
+                <version>1.7.0</version>
+                <configuration>
+                    <from>
+                        <image>openjdk:8u232-jre-stretch</image>
+                    </from>
+                    <to>
+                        <image>docker.io/seata.io/seata</image>
```
Comment (slievrly) [inline server/pom.xml]: seataio/seata-server

Comment (helloworlde) [reply server/pom.xml]: Had modified

**Comment by slievrly** [comment]
Has been discussed with @helloworlde , use profile for configuration isolation, use travis env environment variable to control account.

File: server/pom.xml [hunk 3]
```diff
@@ -153,6 +153,72 @@
...
+                        <tags>
+                            ${image.tags}
+                        </tags>
+                        <auth>
+                            <username>${env.REGISTRY_USERNAME}</username>
+                            <password>${env.REGISTRY_PASSWORD}</password>
+                        </auth>
+                    </to>
+                    <container>
+                        <appRoot>/seata</appRoot>
```
Comment (slievrly) [inline server/pom.xml]: how about seata-server?

File: server/pom.xml [hunk 4]
```diff
@@ -153,6 +153,72 @@
...
+                            <jvmFlag>-XX:+CMSParallelRemarkEnabled</jvmFlag>
+                            <jvmFlag>-XX:+UseCMSInitiatingOccupancyOnly</jvmFlag>
+                            <jvmFlag>-XX:CMSInitiatingOccupancyFraction=75</jvmFlag>
+                            <jvmFlag>-Xloggc:/seata/logs/seata_gc.log</jvmFlag>
+                            <jvmFlag>-verbose:gc</jvmFlag>
+                            <!--netty-->
+                            <jvmFlag>-Dio.netty.leakDetectionLevel=advanced</jvmFlag>
+                        </jvmFlags>
+                        <labels>
+                            <name>Seata</name>
```
Comment (slievrly) [inline server/pom.xml]: seata-server?

File: server/pom.xml [hunk 5]
```diff
@@ -153,6 +153,72 @@
...
+                            ${image.tags}
+                        </tags>
+                        <auth>
+                            <username>${env.REGISTRY_USERNAME}</username>
+                            <password>${env.REGISTRY_PASSWORD}</password>
+                        </auth>
+                    </to>
+                    <container>
+                        <appRoot>/seata</appRoot>
+                        <workingDirectory>/seata</workingDirectory>
```
Comment (slievrly) [inline server/pom.xml]: above

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 1 above)
Comment (zjinlei) [reply server/src/main/java/io/seata/server/ParameterParser.java]: Relying on commons, easy jar version conflict

**Pull request merged**